### PR TITLE
Configure gke-upgrade jobs from 1.0 and 1.1 to master on master

### PR DESCRIPTION
**Pull request by ikehz**
This includes configs for upgrade jobs that drive _from_ `release-1.0` _to_ `master`, and removes old jobs for upgrading from 1.0 on GCE.

### CODE REVIEW

**Comment by ihmccreery** [comment]
The other two parts of this project are in #18028 and #18029.

File: hack/jenkins/e2e.sh [hunk 1]
```diff
@@ -1163,125 +1163,36 @@ case ${JOB_NAME} in
...
-    : ${PROJECT:="kubernetes-jenkins-gke-upgrade"}
-    : ${E2E_UP:="false"}
-    : ${E2E_TEST:="true"}
-    : ${E2E_DOWN:="false"}
-    : ${GINKGO_TEST_ARGS:="--ginkgo.focus=Cluster\sUpgrade.*upgrade-cluster --upgrade-target=${CURRENT_RELEASE_PUBLISHED_VERSION}"}
+  # kubernetes-upgrade-gke-1.1-master
+  #
+  # Configurations for step1, step3, and step5 live in the release-1.1 branch.
+
+  kubernetes-upgrade-gke-1.1-master-step2-upgrade-master)
```
Comment (ixdy) [inline hack/jenkins/e2e.sh]: the PR description suggests you were only changing 1.0, but here you've also changed 1.1... was that intentional?

File: hack/jenkins/e2e.sh [hunk 2]
```diff
@@ -1404,7 +1315,7 @@ case ${JOB_NAME} in
   # 5. runs ci/latest-1.0 e2es again, then
   # 6. runs CURRENT_RELEASE_PUBLISHED_VERSION e2es and tears down the cluster.
 
-    kubernetes-upgrade-1.0-current-release-gce-step1-deploy)
+  kubernetes-upgrade-1.0-current-release-gce-step1-deploy)
```
Comment (ixdy) [inline hack/jenkins/e2e.sh]: why did these not get updated?

File: hack/jenkins/e2e.sh (hunk 1 above)
Comment (ihmccreery) [inline hack/jenkins/e2e.sh]: Oops.  Sorry, changed the PR description.

File: hack/jenkins/e2e.sh (hunk 2 above)
Comment (ihmccreery) [inline hack/jenkins/e2e.sh]: Changed the PR description; only doing GKE right now.

**Comment by ixdy** [comment]
LGTM, but not tagging since you probably want to manually merge

**Comment by ixdy** [comment]
still LGTM, but not tagging to avoid auto-merge

**Comment by ihmccreery** [comment]
Manually merging.

**Comment by mikedanese** [comment]
Squash?

**Comment by ihmccreery** [comment]
Womp.  Sorry about that.

**Pull request merged**